website/src/components/Survey/TrackedTextarea.tsx

```diff
@@ -114,3 +68,49 @@ export const TrackedTextarea = (props: TrackedTextboxProps) => {
     </Stack>
   );
 };
+
+export function getMostProbableLanguage(text: string) {
+  let mostProbableLanguage;
+  try {
+    mostProbableLanguage = LanguageAbbreviations[lande(text)[0][0]];
+  } catch (error) {
+    mostProbableLanguage = "";
+  }
+
+  return mostProbableLanguage;
+}
+
+export function DifferentLanguageModal({ mostProbableLanguage, currentLanguage, setMostProbableLanguage }) {
+  const { isOpen, onOpen, onClose } = useDisclosure();
+  useEffect(() => {
+    if (mostProbableLanguage !== currentLanguage) {
+      onOpen();
+    }
+  }, []);
+
+  return (
+    <>
+      <Modal
+        isOpen={isOpen}
+        onClose={() => {
+          setMostProbableLanguage("");
+          onClose();
+        }}
+        size="xl"
+        scrollBehavior={"inside"}
+      >
+        {/* we kill the event here to disable drag and drop, since it is in the same container */}
+        <ModalOverlay onMouseDown={killEvent}>
+          <ModalContent alignItems="center">
+            <ModalHeader>Switch Language?</ModalHeader>
+            <ModalCloseButton />
+            <ModalBody>
+              Do you want to switch language? The detected language is <b>{mostProbableLanguage}</b>, whereas your
```

Comment:
we should change this language to be something like

`We've detected this content to be in {mostProbableLanguage} while your chosen language is {currentLanguage}.  Please write your answer in {currentLanguage} or go back to the <link>dashboard</link> and change your language.`

Reply:
I didn't know that a good solution was already implemented (forgot to sync 😅)
![image](https://user-images.githubusercontent.com/113536090/217469662-109d3db8-8fe2-4204-8e2c-65ab472fc659.png)
If you still want to show the modal on review, then I'll do it. 
If not then I think this should be closed

Reply:
If you think the changes that got merged already are sufficient, I'll let you close this out.  Otherwise, I'll review again after the merge conflicts are resolved.

Either way, thank you for putting something together for this problem.

Reply:
I would only change the progress bar to red in case of conflicting languages, as I think there could be some people who'll miss the letters on the corner. Apart from this I think the implemented solution is great.